header-rewrite-filter/header_rewrite.cc

```diff
@@ -42,15 +42,16 @@ HttpHeaderRewriteFilter::HttpHeaderRewriteFilter(HttpHeaderRewriteFilterConfigSh
     }
 
     // determine if it's request/response
-    if (tokens.at(0) != Utility::HTTP_REQUEST && tokens.at(0) != Utility::HTTP_RESPONSE) {
-      fail("first argument must be <http-response/http-request>");
+    if (tokens.at(0) != Utility::HTTP_REQUEST && tokens.at(0) != Utility::HTTP_RESPONSE && tokens.at(0) != Utility::HTTP_REQUEST_RESPONSE) {
+      fail("first argument must be <http-response/http-request/http>");
       setError();
       return;
     }
     const bool isRequest = (tokens.at(0) == Utility::HTTP_REQUEST);
+    const bool isResponse = (tokens.at(0) == Utility::HTTP_RESPONSE);
```

Comment:
Do these need to also check `HTTP_REQUEST_RESPONSE`?

Reply:
Right now, the only processor that is `HTTP_REQUEST_RESPONSE` is the Set Bool Processor, which is inserted into its data structure within the switch statement, so only http-request and http-response processors need to be added to their respective data structures using the above checks.